core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -1540,6 +1574,17 @@ public Map<String, Map<String, Long>> getConnectorRequestsCounts(
       updateMap(statusCountsMap, connectorRequestsStatusObj);
     }
 
+    if (RequestMode.MY_APPROVALS == requestMode) {
+      // Make sure to remove any requests which the requestor can not approve
+      List<Object[]> connectorApprovalStatusObj =
```

Comment:
why not to replace with count query?
Count query is much faster since it doesn't need to load all the field values